### ISSUE TITLE
[BUG] Wrong initialisation of TWE distance in _twe_numba.py

### ISSUE THREAD

**Issue by irazza** [alan-turing-institute/sktime#6182]
**Describe the bug**
 When using the banded version in the twe_distance function with parameter window < 1.0, the current initialization of the cost matrix is incorrect. It is now filled with all zeros except the first row and column, which are filled with +inf (with the exception of the cell 0,0). This should be instead filled with +inf.

This leads to an incorrect output of the TWE distance because when computing a cell of the matrix, the algorithm takes as a possibility, i.e. cost_matrix[.,.], a value of 0 in a neighboring cell outside the band instead of +inf. 

**To Reproduce**
Given two lists: [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0], [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0], and window=0.1; the returned distance is 4.0009, while it should be 86. Due to this window constraint, it can only use the value on the diagonal, choosing to produce a mismatch for each entry. 

**Expected behavior**

The initialization of the cost matrix should be changed to 
(See rows 82 to 85)

cost_matrix = np.full((x_size, y_size), np.inf)
// cost_matrix[0, 1:] = np.inf
// cost_matrix[1:, 0] = np.inf
cost_matrix[0, 0] = 0

**Additional context**

**Versions**
<details>

System:
    python: 3.12.0 (v3.12.0:0fb18b02c8, Oct  2 2023, 09:45:56) [Clang 13.0.0 (clang-1300.0.29.30)]
executable: /path
   machine: macOS-14.4-arm64-arm-64bit

Python dependencies:
          pip: 24.0
       sktime: 0.27.0
      sklearn: 1.4.1.post1
       skbase: 0.7.5
        numpy: 1.26.4
        scipy: 1.12.0
       pandas: 2.1.4
   matplotlib: 3.8.3
       joblib: 1.3.2
        numba: 0.59.0
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

</details>

**Comment by fkiraly** [comment]
I see - I think you are right. Just to clarify, you think it should be filled with +inf *outside the permitted band*, right?

When we fix this, we should:

* also update the docstrings to be very specific about the output matrix
* ensure to add a test that checks actual against expected for `window < 1.0`

Further questions:

* TWE distance is a "standard" distance afaik, right? does this mean, it would lead to spurious/incorrect low performance of baselines in predicive performance benchmarking studies?
* can you explain how you obtain 86? I get 44, summing up the Euclidean distances.

**Comment by fkiraly** [comment]
PS, isn't the following code simpler and shorter, and equivalent?

```python
cost_matrix[:, :] = np.inf
cost_matrix[0, 0] = 0.
```

**Comment by albertoazzari** [comment]
Yes, exactly outside the permitted band, we should fill the bounding matrix with +inf.

1. I think that this bug could lead to incorrect distance measures, in any case, it can result in spurious analyses.
2. Following the rules to compute the costs, if you have to compute the values in the diagonal cells, you should sum the Euclidean distances both for i,j and i-1, j-1. 
`  x1 = np.array([1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0])
    x2 = np.array([10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0])
    x1_p = np.insert(x1, 0, 0)
    x2_p = np.insert(x2, 0, 0)
    dist = 0
    for i in range(1, len(x1_p)):
        dist += abs(x1_p[i] - x2_p[i]) + abs(x1_p[i-1] - x2_p[i-1])
    print(dist)`
    dist = 86.0
3. I don't think z-scoring the time series in input is mandatory.

**Comment by fkiraly** [comment]
Ah, thanks for the clarification. I forgot the i-1 terms.

Would you like to make a pull request with a fix?

**Comment by fkiraly** [comment]
> I don't think z-scoring the time series in input is mandatory.

Yes, of course - my issue is that it is not actually clear whether it is done or not, to a user.
